### PR TITLE
Extra options for poloidal plots

### PR DESCRIPTION
**Pull request by johnomotani**
Split out 2d-plot changes from #38.

### CODE REVIEW

File: xbout/plotting/plotfuncs.py [hunk 1]
```diff
@@ -169,6 +199,37 @@ def plot2d_wrapper(da, method, *, ax=None, separatrix=True, targets=True,
         extend = kwargs.get('extend', 'neither')
         fig.colorbar(artists[0], ax=ax, extend=extend)
 
+    if gridlines is not None:
+        # convert gridlines to dict
+        if not isinstance(gridlines, dict):
+            gridlines = {'x': gridlines, 'y': gridlines}
+
+        for key in gridlines:
+            value = gridlines[key]
```
Comment (TomNicholas) [inline xbout/plotting/plotfuncs.py]:
```python
for key, value in gridlines
```

File: xbout/plotting/plotfuncs.py [hunk 2]
```diff
@@ -169,6 +199,37 @@ def plot2d_wrapper(da, method, *, ax=None, separatrix=True, targets=True,
...
+            if value is True:
+                gridlines[key] = slice(None)
+            elif isinstance(value, int):
+                gridlines[key] = slice(0, None, value)
+            elif value is not None:
+                if not isinstance(value, slice):
+                    raise ValueError('Argument passed to gridlines must be bool, int or '
+                                     'slice. Got ' + str(value))
```
Comment (TomNicholas) [inline xbout/plotting/plotfuncs.py]: use ```type(value)``` instead of `str(value)`

File: xbout/plotting/plotfuncs.py [hunk 3]
```diff
@@ -169,6 +199,37 @@ def plot2d_wrapper(da, method, *, ax=None, separatrix=True, targets=True,
...
+
+        Z_global = da['Z']
+        Z_global.attrs['metadata'] = da.metadata
+
+        R_regions = _decompose_regions(da['R'])
+        Z_regions = _decompose_regions(da['Z'])
+
+        for R, Z in zip(R_regions, Z_regions):
+            if 'x' in gridlines and gridlines['x'] is not None:
+                plt.plot(R[:, gridlines['y']], Z[:, gridlines['y']], color='k', lw=0.1)
```
Comment (TomNicholas) [inline xbout/plotting/plotfuncs.py]: `R.isel(y=gridlines['y'])` is slightly clearer and more robust imo

**Comment by johnomotani** [comment]
sorry, gridlines doesn't work at the moment. will debug after the meeting

File: xbout/plotting/plotfuncs.py [hunk 4]
```diff
@@ -169,6 +199,50 @@ def plot2d_wrapper(da, method, *, ax=None, separatrix=True, targets=True,
...
+        R_regions = _decompose_regions(da['R'])
+        Z_regions = _decompose_regions(da['Z'])
+
+        for R, Z in zip(R_regions, Z_regions):
+            if (not da.metadata['bout_xdim'] in R.dims
+                    and not da.metadata['bout_ydim'] in R.dims):
+                # Small regions around X-point do not have segments in x- or y-directions,
+                # so skip
+                continue
+            if 'x' in gridlines and gridlines['x'] is not None:
```
Comment (TomNicholas) [inline xbout/plotting/plotfuncs.py]: You can just do `if gridlines.get('x'):` if you want, the default return value of `.get()` if the key isn't found is `None`.

**Pull request merged**